packages/react/src/pill/Pill.tsx

```diff
@@ -0,0 +1,53 @@
+import { type KeyboardEvent, type ReactNode, forwardRef } from "react";
+
+import { type BoxProps } from "../box";
+import { Button } from "../button";
+import { Text } from "../text";
+import * as styles from "./Pill.css";
+
+type PillProps = BoxProps<
+  "button",
+  {
+    endDecorator?: ReactNode;
+    onRemove?: () => void;
+    startDecorator?: ReactNode;
+  } & styles.PillVariants
+>;
+
+export const Pill = forwardRef<HTMLButtonElement, PillProps>(
+  (
+    {
+      children,
+      className,
+      endDecorator,
+      onRemove,
+      size = "md",
+      startDecorator,
+      ...props
+    },
+    ref,
+  ) => {
+    const handleKeyDown = (e: KeyboardEvent<HTMLButtonElement>) => {
+      if (e.key === "Backspace" && onRemove) {
+        onRemove();
+      }
+    };
+
+    return (
+      <Button
+        endDecorator={endDecorator}
+        onKeyDown={handleKeyDown}
+        ref={ref}
+        startDecorator={startDecorator}
+        {...styles.pill({ size }, className)}
+        {...props}
+      >
+        <Text display="block" fontSize="sm" truncate>
+          {children}
+        </Text>
```

Comment:
Lets use tooltip as theres no way to see truncated text.

```suggestion
        <Tooltip auto content={children}>
          <Text display="block" fontSize="sm" truncate>
            {children}
          </Text>
        </Tooltip>
```
With auto, it will only be shown text is truncated.